examples/webgpu_postprocessing_dof.html

```diff
@@ -0,0 +1,218 @@
+
+							const x = 200 * ( i - xgrid / 2 );
+							const y = 200 * ( j - ygrid / 2 );
+							const z = 200 * ( k - zgrid / 2 );
+
+							mesh.setMatrixAt( index, matrix.identity().setPosition( x, y, z ) );
+							mesh.setColorAt( index, color );
+							index ++;
+
+						}
+
+					}
+
+				}
+
+				// renderer
+
+				renderer = new WebGPURenderer( { antialias: true } );
+				renderer.setPixelRatio( window.devicePixelRatio );
+				renderer.setSize( window.innerWidth, window.innerHeight );
+				renderer.setAnimationLoop( animate );
+				document.body.appendChild( renderer.domElement );
+
+				// post processing
+
+				postProcessing = new PostProcessing( renderer );
+
+				const scenePass = pass( scene, camera );
+
+				const scenePassColor = scenePass.getTextureNode();
+				const scenePassViewZ = scenePass.getViewZNode();
+
+				const dofPass = scenePassColor.dof( scenePassViewZ );
+
+				postProcessing.outputNode = dofPass;
+
+				// controls
+
+				renderer.domElement.style.touchAction = 'none';
+				renderer.domElement.addEventListener( 'pointermove', onPointerMove );
+
+				window.addEventListener( 'resize', onWindowResize );
+
+				// stats
+
+				stats = new Stats();
+				document.body.appendChild( stats.dom );
+
+				// gui
+
+				const effectController = {
+
+					focus: 500.0,
+					aperture: 5,
+					maxblur: 0.01
+
+				};
+
+				function updateEffect( ) {
+
+					dofPass.focus.value = effectController.focus;
+					dofPass.aperture.value = effectController.aperture * 0.00001;
+					dofPass.maxblur.value = effectController.maxblur;
+
+				}
+
+				const gui = new GUI();
+				gui.add( effectController, 'focus', 10.0, 3000.0, 10 ).onChange( updateEffect );
+				gui.add( effectController, 'aperture', 0, 10, 0.1 ).onChange( updateEffect );
+				gui.add( effectController, 'maxblur', 0.0, 0.01, 0.001 ).onChange( updateEffect );
+
+				updateEffect();
+
+			}
+
+			function onPointerMove( event ) {
+
+				if ( event.isPrimary === false ) return;
+
+				mouseX = event.clientX - windowHalfX;
+				mouseY = event.clientY - windowHalfY;
+
+			}
+
+			function onWindowResize() {
+
+				windowHalfX = window.innerWidth / 2;
+				windowHalfY = window.innerHeight / 2;
+
+				width = window.innerWidth;
+				height = window.innerHeight;
+
+				camera.aspect = width / height;
+				camera.updateProjectionMatrix();
+
+				renderer.setSize( width, height );
+
+			}
+
+			function animate() {
+
+				render();
+				stats.update();
+
+			}
+
+			function render() {
+
+				const time = Date.now() * 0.00005;
+
+				camera.position.x += ( mouseX - camera.position.x ) * 0.036;
+				camera.position.y += ( - ( mouseY ) - camera.position.y ) * 0.036;
+
+				camera.lookAt( scene.position );
+
+				for ( let i = 0; i < mesh.count; i ++ ) {
+
+					const h = ( 360 * ( i / mesh.count + time ) % 360 ) / 360;
+					color.setHSL( h, 1, 0.5 );
+					mesh.setColorAt( i, color );
+
+				}
+
+				mesh.instanceColor.needsUpdate = true;
```

Comment:
@RenaudRohlinger I originally wanted to use `THREE.DynamicDrawUsage` for the colors but the colors did not get updated during rendering. Even manually setting `mesh.instanceColor.needsUpdate = true;` did not help.

As a workaround, I had to fall back to `THREE.StaticDrawUsage` with manual updates but for this demo it would be better to use `THREE.DynamicDrawUsage`. Maybe related: #28716

Reply:
Seems like this issue was always there, the usage is not properly forwarded to the new buffer in InstanceNode. I will have a look.

Reply:
The issue was in BufferAttributeNode. the setUsage method wasn't handling so the buffer was never init with dynamic usage.

Fixed in https://github.com/mrdoob/three.js/pull/28742.